src/api/me.py

```diff
@@ -634,3 +639,106 @@ async def add_user_to_following(
             UserAlreadyInFollowingException()
     else:
         raise UserAlreadyInFollowingException()
+
+
+@router.post(
+    '/reviews/{alcohol_id}',
+    response_class=Response,
+    status_code=status.HTTP_201_CREATED,
+    summary='Create a review'
+)
+async def create_review(
+        alcohol_id: str,
+        review_create_payload: ReviewCreate,
+        current_user: UserDb = Depends(get_valid_user),
+        db: Database = Depends(get_db)
+):
+    if not await ReviewDatabaseHandler.validate_rating(review_create_payload.rating):
+        raise WrongRatingFormat()
+
+    if await ReviewDatabaseHandler.check_if_review_exists(
+            db.reviews,
+            alcohol_id,
+            str(current_user['_id'])):
+        raise ReviewAlreadyExists()
+
+    if await ReviewDatabaseHandler.create_review(
+            db.reviews,
+            str(current_user['_id']),
+            alcohol_id,
+            str(current_user['username']),
+            review_create_payload
+    ):
+        await ReviewDatabaseHandler.add_rating_to_alcohol(
+            db.alcohols,
+            alcohol_id,
+            review_create_payload.rating
+        )
+
+
+@router.delete(
+    path='/reviews/{review_id}',
+    status_code=status.HTTP_204_NO_CONTENT,
+    summary='Delete your review'
+)
+async def delete_review(
+        review_id: str,
+        current_user: UserDb = Depends(get_valid_user),
+        db: Database = Depends(get_db)
+) -> None:
+    """
+    Delete your review by id
+    """
+    if not await ReviewDatabaseHandler.check_if_review_belongs_to_user(
+            db.reviews,
+            review_id,
+            str(current_user['_id'])):
+        raise ReviewDoesNotBelongToUser()
+
+    alcohol_id = await ReviewDatabaseHandler.get_alcohol_id(db.reviews, review_id)
+    rating = await ReviewDatabaseHandler.get_rating(db.reviews, review_id)
+
+    if await ReviewDatabaseHandler.delete_review(db.reviews, review_id):
+        await ReviewDatabaseHandler.remove_rating_from_alcohol(db.alcohols, alcohol_id, rating)
+
+
+@router.put(
+    path='/reviews/{review_id}',
+    response_model=Review,
+    status_code=status.HTTP_200_OK,
+    summary='Update your review'
+)
+async def update_review(
+        review_id: str,
+        review_update_payload: ReviewUpdate,
+        current_user: UserDb = Depends(get_valid_user),
+        db: Database = Depends(get_db)
+):
+    if not await ReviewDatabaseHandler.validate_rating(review_update_payload.rating):
+        raise WrongRatingFormat()
+
+    if not await ReviewDatabaseHandler.check_if_review_exists_by_id(
+            db.reviews,
+            review_id,
+    ):
+        raise ReviewNotFound()
+
+    if not await ReviewDatabaseHandler.check_if_review_belongs_to_user(
+            db.reviews,
+            review_id,
+            str(current_user['_id'])):
+        raise ReviewDoesNotBelongToUser()
+
+    alcohol_id = await ReviewDatabaseHandler.get_alcohol_id(db.reviews, review_id)
```

Comment:
Zastanawiam się, czy na dobrą sprawę nie możemy wymusić czegoś takiego:
`/reviews/{review_id}/alcohol/{alcohol_id}`.
Wtedy nie trzeba będzie szukać id alkoholu w bazie. A oni na froncie to powinni mieć